docs/examples/move/nft_marketplace/sources/rental_extension.move

```diff
@@ -0,0 +1,417 @@
+    /// Even for creators that do not wish to enforce royalties. Provides authorized access to an
+    /// empty TransferPolicy.
+    public struct ProtectedTP<phantom T> has key, store {
+        id: UID,
+        transfer_policy: TransferPolicy<T>,
+        policy_cap: TransferPolicyCap<T>
+    }
+
+    // === Public Functions ===
+
+    /// Enables someone to install the Marketplace extension in their Kiosk.
+    public fun install(
+        kiosk: &mut Kiosk,
+        cap: &KioskOwnerCap,
+        ctx: &mut TxContext,
+    ) {
+        kiosk_extension::add(Rental {}, kiosk, cap, PERMISSIONS, ctx);
+    }
+
+     /// Remove the extension from the Kiosk. Can only be performed by the owner,
+    /// The extension storage must be empty for the transaction to succeed.
+    public fun remove(kiosk: &mut Kiosk, cap: &KioskOwnerCap, _ctx: &mut TxContext) {
+        kiosk_extension::remove<Rental>(kiosk, cap);
+    }
+
+    /// Mints and shares a ProtectedTP & a RentalPolicy object for type T.
+    /// Can only be performed by the publisher of type T.
+    public fun setup_renting<T>(publisher: &Publisher, amount_bp: u64, ctx: &mut TxContext) {
+        // Creates an empty TP and shares a ProtectedTP<T> object.
+        // This can be used to bypass the lock rule under specific conditions.
+        // Storing inside the cap the ProtectedTP with no way to access it
+        // as we do not want to modify this policy
+        let (transfer_policy, policy_cap) = transfer_policy::new<T>(publisher, ctx);
+
+        let protected_tp = ProtectedTP {
+            id: object::new(ctx),
+            transfer_policy,
+            policy_cap,
+        };
+
+        let rental_policy = RentalPolicy<T> {
+            id: object::new(ctx),
+            balance: balance::zero<IOTA>(),
+            amount_bp,
+        };
+
+        transfer::share_object(protected_tp);
+        transfer::share_object(rental_policy);
+    }
+
+    /// Enables someone to list an asset within the Marketplace extension's Bag,
+    /// creating a Bag entry with the asset's ID as the key and a Rentable wrapper object as the value.
+    /// Requires the existance of a ProtectedTP which can only be created by the creator of type T.
+    /// Assumes item is already placed (& optionally locked) in a Kiosk.
+    public fun list<T: key + store>(
+        kiosk: &mut Kiosk,
+        cap: &KioskOwnerCap,
+        protected_tp: &ProtectedTP<T>,
+        item_id: ID,
+        duration: u64,
+        price_per_day: u64,
+        ctx: &mut TxContext,
+    ) {
+        assert!(kiosk_extension::is_installed<Rental>(kiosk), EExtensionNotInstalled);
+
+        kiosk.set_owner(cap, ctx);
+        kiosk.list<T>(cap, item_id, 0);
+
+        let coin = coin::zero<IOTA>(ctx);
+        let (object, request) = kiosk.purchase<T>(item_id, coin);
+
+        let (_item, _paid, _from) = protected_tp.transfer_policy.confirm_request(request);
+
+        let rentable = Rentable {
```

Comment:
Check that price_per_day * duration <= MAX_VALUE_U64

Reply:
It’s checking in `rent` function

  ```
     let max_price_per_day = MAX_VALUE_U64 / rentable.duration;
        assert!(rentable.price_per_day <= max_price_per_day, ETotalPriceOverflow);
```

Do we have to check it twice?

Reply:
Ah I see your comment below

Reply:
Done